cpp/src/DataStorm/SessionI.cpp

```diff
@@ -1134,15 +1138,14 @@ SessionI::runWithTopics(
     for (auto topic : topics)
     {
         retained.push_back(topic);
-        unique_lock<mutex> l(topic->getMutex());
+        unique_lock<mutex> lock(topic->getMutex());
         if (topic->isDestroyed())
         {
             continue;
         }
-        _topicLock = &l;
+        _topicLock = &lock;
         fn(topic);
         _topicLock = nullptr;
-        l.unlock();
```

Comment:
No need for the explicit unlock here, the lock is being destroyed as it is about to go out of scope.